### PR TITLE
feat(ChainSelect): wip: main functions of ChainSelect 

### PR DESCRIPTION
**Pull request by zctocm**
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/rigel/31)

### CODE REVIEW

File: src/pages/Shuttle/index.js [hunk 1]
```diff
@@ -1,5 +1,61 @@
...
+  useEffect(() => {
+    if (fromChain === toChain) {
+      setFromChain(prevToChain)
+      setToChain(prevFromChain)
+    } else {
+      if (fromChain !== ChainShortNameCfx) {
+        setToChain(DefaultToChain)
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (angelia-yuqi-personal) [inline src/pages/Shuttle/index.js:30]: why use react-hooks/exhaustive-deps, may need to add `setFromChain` and `setToChain`

File: src/pages/components/ChainSelect/index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+export default ChainSelect
+
+/**
+ * get the data that will show in the Chain Dropdown list
+ * @param {*} type
+ * @param {*} chain
+ * @param {*} fromChain
+ */
+function getChainsData(type, chain, fromChain) {
```
Comment (angelia-yuqi-personal) [inline src/pages/components/ChainSelect/index.js:62]: can use a data hook

**Review by angelia-yuqi-personal — CHANGES_REQUESTED** [review]

File: src/components/MenuItem/index.js [hunk 3]
```diff
@@ -1,21 +1,36 @@
 import PropTypes from 'prop-types'
-import {selectedIcon as defaultSelectedIcon} from '../../assets/images'
-
-function MenuItem({children, itemKey, selected, selectedIcon, icon, onClick}) {
+import {
+  selectedIcon as defaultSelectedIcon,
+  disabledIcon,
+} from '../../assets/images'
```
[the remark this answers is not in the document]
Comment (zctocm) [reply src/components/MenuItem/index.js:5]: done

File: src/pages/Shuttle/index.js (hunk 1 above)
Comment (zctocm) [reply src/pages/Shuttle/index.js]: done

File: src/pages/components/ChainSelect/index.js (hunk 2 above)
Comment (zctocm) [reply src/pages/components/ChainSelect/index.js]: done

**Review by angelia-yuqi-personal — APPROVED** [review]

**Pull request merged**